### PR TITLE
Feature/projected data in charts

### PR DESCRIPTION
**Pull request by Bluesmile82**
This PR adds the projected data and dividing line with labels for the line and stacked-area type graphs:

image

image

### CODE REVIEW

File: src/components/charts/chart/chart.md [hunk 1]
```diff
@@ -100,6 +100,62 @@ const toggleChartType = (filtersSelected) => {
...
+}
+<React.Fragment>
+  <Button onClick={toggleChartType} >
+    Toggle type
+  </Button>
+  <Chart
+    type={state.type}
+    config={state.config}
+    data={state.data}
+    projectedData={state.projectedData}
```
Comment (weberjavi) [inline src/components/charts/chart/chart.md]: I know this `prop` is passed on `Chart` but consumed by a child component (`Line` or `Area` charts), but would be great have it documented in the parent component as is the one that we are going to be using on each project.
Knowing the `shape` of the `data` and `config` of `projectedData` without needing to dive into the code would be great.
In the example this are the PropTypes:
```js
projectedData: PropTypes.shape({
  data: PropTypes.arrayOf(PropTypes.shape({
    x: PropTypes.number,
    y: .oneOfType([ PropTypes.number, PropTypes.arrayOf(PropTypes.number) ])
    label: PropTypes.string,
    isRange:PropTypes.bool
  })),
  config: PropTypes.shape({
    dividerLine: PropTypes.shape({
      pre: PropTypes.string,
      post: PropTypes.string
    })
  })
})
```

**Comment by Bluesmile82** [comment]
Ready! Also the legend has been added. Please check again

File: src/components/charts/projected-data/divider-line.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const DividerLine = ({ labels, x }) => (
+  <ReferenceLine x={x} isFront={false}>
+    <Label
+      position="top"
+      strokeWidth={0.5}
+      content={content => (
+        <g fill="#b1b1c1" style={{ fontSize: '0.9rem' }}>
+          <text x={content.viewBox.x - 140} y="35">
+            {labels && labels.pre || 'Historical emissions'}
```
[the remark this answers is not in the document]
Comment (aabdaab) [reply src/components/charts/projected-data/divider-line.js]: this will be passed with the config anyway (in the project)

**Comment by aabdaab** [comment]
works great! I'd just make small changes to the tooltip 
image
^ could we increase the margin a bit?
And maybe let's decrease min-height for the tagsContainer to 60px or so
image

**Comment by Bluesmile82** [comment]
Thanks! Ready again

**Pull request merged**